frontend/src/metabase/dashboard/components/Dashboard.jsx

```diff
@@ -228,6 +230,10 @@ export default class Dashboard extends Component {
     });
   };
 
+  onCancel = () => {
+    this.setState({ isSharing: false });
```

Comment:
It's not part of this PR, but `isSharing` might make more sense in Redux rather than component state. I think it hinges on whether we want to tie the sidebar behavior in with any other UI changes that are managed in redux. e.g. if we the sharing sidebar to auto-close when switching to editing mode, we can more cleanly handle that with reducers.

Reply:
@paulrosenzweig This sounds really appealing to me—there's already some funky behavior surrounding getting things closed. I spent a while exploring the actions, `Dashboard` state, etc. and didn't make too much headway. Do you have 10-15 minutes in the next few days to talk through this a little more?

Reply:
Ahhh cool I'd like a better explanation of this too - can I join the talk about it? I was looking at how other things were done like this using `props` instead of `state` but I couldn't figure out how that worked since `props` are supposed to be immutable?

There's clearly a redux-ism here I'm missing!